.github/workflows/prototype-tests.yml

```
@@ -21,7 +21,31 @@ jobs:
         with:
           python-version: 3.7
 
-      - name: Upgrade system packages
+      - name: Generate cache params
+        id: cache-params
+        shell: python
+        run: |
+          import datetime
+          import pathlib
+          import sys
+          
+          cache_path = pathlib.Path(sys.executable).parents[1]
+          print(cache_path)
+          print(f"::set-output name=path::{cache_path}")
+          
+          today = datetime.datetime.utcnow()
+          yesterday = today - datetime.timedelta(1)
+          cache_date = f"{today if today.hour > 10 else yesterday:%Y%m%d}"
```

This makes sure we cache the current nightly builds correctly. We will have a "constant" env every day starting from UTC+0 11a:00.

---

Actually, `torchdata` only starts their nightly workflows at [UTC+0 11:00](https://github.com/pytorch/data/blob/92d18b088eb43b9805bed5c90a0afca87292a338/.github/workflows/nightly_release.yml#L6). Thus, we should only cache after they are done:

```suggestion
          cache_date = f"{today if today.hour > 11 else yesterday:%Y%m%d}"
```